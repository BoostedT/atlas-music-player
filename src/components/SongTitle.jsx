import React from "react";

export default function SongTitle({
  title = "Unknown Title",
  artist = "Unknown Artist",
}) {
  return (
    <div className="text-center">
      <h3 className="text-base font-semibold leading-tight truncate">{title}</h3>
      <p className="text-sm text-gray-500 leading-snug truncate">{artist}</p>
    </div>
  );
}
