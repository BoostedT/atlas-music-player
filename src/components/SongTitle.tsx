import React from "react";

interface Song {
  id: string;
  title: string;
  artist: string;
  duration: number; // seconds
  artworkUrl: string;
  lyrics?: string;
}

interface Props {
  song: Song;
}

export default function SongTitle({ song }: Props) {
  return (
    <div>
      <h2 className="text-lg font-bold">{song.title}</h2>
      <p className="text-sm text-gray-600">{song.artist}</p>
    </div>
  );
}
