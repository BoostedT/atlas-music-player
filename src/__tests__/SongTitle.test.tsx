import { render } from "@testing-library/react";
import React from "react";
import SongTitle from "../components/SongTitle";

describe("SongTitle", () => {
  it("renders with only title", () => {
    const { asFragment } = render(
      <SongTitle song={{ title: "Test Song", artist: "" }} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders with title and artist", () => {
    const { asFragment } = render(
      <SongTitle song={{ title: "Test Song", artist: "Test Artist" }} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders with long title and artist", () => {
    const { asFragment } = render(
      <SongTitle
        song={{
          title: "A Very Very Long Song Title That Should Wrap",
          artist: "An Incredibly Long Artist Name To Test Layout",
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
