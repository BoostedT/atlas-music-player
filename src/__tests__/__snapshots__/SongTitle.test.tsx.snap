// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle > renders with long title and artist 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="text-lg font-bold"
    >
      A Very Very Long Song Title That Should Wrap
    </h2>
    <p
      class="text-sm text-gray-600"
    >
      An Incredibly Long Artist Name To Test Layout
    </p>
  </div>
</DocumentFragment>
`;

exports[`SongTitle > renders with only title 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="text-lg font-bold"
    >
      Test Song
    </h2>
    <p
      class="text-sm text-gray-600"
    />
  </div>
</DocumentFragment>
`;

exports[`SongTitle > renders with title and artist 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="text-lg font-bold"
    >
      Test Song
    </h2>
    <p
      class="text-sm text-gray-600"
    >
      Test Artist
    </p>
  </div>
</DocumentFragment>
`;
